source/extensions/filters/network/dubbo_proxy/router/route_matcher.cc

```diff
@@ -0,0 +1,276 @@
+    break;
+  }
+}
+
+MethodRouteEntryImpl::MethodRouteEntryImpl(
+    const envoy::config::filter::network::dubbo_proxy::v2alpha1::Route& route)
+    : RouteEntryImplBase(route), method_name_(route.match().method().name()),
+      is_contain_wildcard_(Utility::isContainWildcard(method_name_)) {
+  if (route.match().method().params_match_size() != 0) {
+    parameter_route_ = std::make_shared<ParameterRouteEntryImpl>(route);
+  }
+  ENVOY_LOG(debug, "dubbo route matcher: method name {}", method_name_);
+}
+
+MethodRouteEntryImpl::~MethodRouteEntryImpl() {}
+
+RouteConstSharedPtr MethodRouteEntryImpl::matches(const MessageMetadata& metadata,
+                                                  uint64_t random_value) const {
+  if (metadata.hasHeaders() && !RouteEntryImplBase::headersMatch(metadata.headers())) {
+    ENVOY_LOG(error, "dubbo route matcher: headers not match");
+    return nullptr;
+  }
+  ENVOY_LOG(debug, "dubbo route matcher: method name match");
+  if (!metadata.method_name().has_value()) {
+    ENVOY_LOG(error, "dubbo route matcher: there is no method name in the metadata");
+    return nullptr;
+  }
+
+  if (method_name_ == "*") {
+    return clusterEntry(random_value);
+  }
+
+  if (is_contain_wildcard_) {
+    if (!Utility::wildcardMatch(metadata.method_name().value().c_str(), method_name_.c_str())) {
+      ENVOY_LOG(
+          debug,
+          "dubbo route matcher: method matching failed, origin method '{}', input method '{}'",
+          method_name_, metadata.method_name().value());
+      return nullptr;
+    }
+  } else {
+    if (metadata.method_name() != method_name_) {
+      ENVOY_LOG(
+          debug,
+          "dubbo route matcher: method matching failed, origin method '{}', input method '{}'",
+          method_name_, metadata.method_name().value());
+      return nullptr;
+    }
+  }
+
+  if (parameter_route_.has_value()) {
+    ENVOY_LOG(debug, "dubbo route matcher: parameter matching is required");
+    return parameter_route_.value()->matches(metadata, random_value);
+  }
+
+  return clusterEntry(random_value);
+}
+
+RouteMatcher::RouteMatcher(const RouteConfig& config)
+    : interface_name_(config.interface()), group_(config.group()), version_(config.version()) {
+  using envoy::config::filter::network::dubbo_proxy::v2alpha1::RouteMatch;
+
+  for (const auto& route : config.routes()) {
+    switch (route.match().match_specifier_case()) {
+    case RouteMatch::MatchSpecifierCase::kMethod:
+      routes_.emplace_back(new MethodRouteEntryImpl(route));
+      ENVOY_LOG(debug, "dubbo route matcher: create the method route entry");
+      break;
+    default:
+      NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
+    }
+  }
+  ENVOY_LOG(debug, "dubbo route matcher: routes list size {}", routes_.size());
+}
+
+RouteConstSharedPtr RouteMatcher::route(const MessageMetadata& metadata,
+                                        uint64_t random_value) const {
+  if (interface_name_ == metadata.service_name() &&
```

Comment:
Can interface_name and service_name be unified?

Reply:
It has been uniformly named service name.